### DISCUSSION TITLE
Chen et al. 2021 "Transunet: Transformers make strong encoders for medical image segmentation."

### DISCUSSION THREAD

**Discussion by mxochicale** [mxochicale/unetvit4sclera#4] [References]
* Chen, Jieneng, Yongyi Lu, Qihang Yu, Xiangde Luo, Ehsan Adeli, Yan Wang, Le Lu, Alan L. Yuille, and Yuyin Zhou. **"Transunet: Transformers make strong encoders for medical image segmentation."** arXiv preprint arXiv:2102.04306 (2021).
  * [google-citations](https://scholar.google.com/scholar?cites=1117067442557737826&as_sdt=2005&sciodt=0,5&hl=en)
  * [arxiv](https://arxiv.org/pdf/2102.04306v1)
  * [GitHub-project](https://github.com/Beckschen/TransUNet)
  * Videos: https://www.youtube.com/watch?v=jKBJITQ8xJY
  * **Implementation Details**: 
    * All Transformer backbones (i.e.,ViT) and ResNet-50 (denoted as “R-50”) were **pretrained** on ImageNet. 
    * Models are trained with SGD optimizer with learning rate 0.01, momentum 0.9 and weight decay 1e-4.
    * The default batch size is 24 and the default number of training iterations are 20k for ACDC dataset and 14k for Synapse dataset respectively.
    * The default batch size is 24 and the default number of training iterations are 20k for ACDC dataset and 14k for Synapse dataset respectively.
  * **Ablation study** : 
    * 1) the number of skip-connections (0-skip, 1-skip, 3-skip); 
      *  "We can see that adding more skip-connections generally leads to a better segmentation performance."
    * 2) input resolution; 
      * "As [4] indicated, increasing the effective sequence length shows robust improvements. For TransUNet, changing the resolution scale from 224×224 to 512×512 results in 6.88% improvement in average DSC, at the expense of a much larger computational cost."
    * 3) sequence length and patch size and 
      * "It is observed that a higher segmentation performance is usually obtained with smaller patch size. Note that the Transformer’s sequence length is inversely proportional to the square of the patch size"
    * 4) model scaling (Base DSC %77.48; Larger DSC %78.52)
      * “base” model, the hidden size D=12, number of layers=768, MLP size=3072, and number of heads=12
      * “large” model, the hidden size D=24, number of layers=1024, MLP size=4096, and number of heads=16
